admin/src/lib.rs

```diff
@@ -0,0 +1,323 @@
+	let nb_column = cli.shared().nb_columns.unwrap_or(1);
+	let metadata_path = cli.shared().use_metadata.clone().unwrap_or_else(|| {
+		let mut p = db_path.clone();
+		p.push("metadata");
+		p
+	});
+	let mut options = if let Some(metadata) = parity_db::Options::load_metadata(&metadata_path)
+		.map_err(|e| format!("Error resolving metas: {:?}", e))? {
+		let mut options = parity_db::Options::with_columns(db_path.as_path(), 0);
+		options.columns = metadata.columns;
+		options.salt = metadata.salt;
+		options
+	} else {
+		parity_db::Options::with_columns(db_path.as_path(), nb_column)
+	};
+	options.sync_wal = !cli.shared().no_sync;
+	options.sync_data = !cli.shared().no_sync;
+	options.stats = cli.shared().with_stats;
+	println!("Options {:?}, {:?}", cli, options);
+	match cli.subcommand {
+		SubCommand::Stats(stat) => {
+			let db = parity_db::Db::open_read_only(&options)
+				.map_err(|e| format!("Invalid db: {:?}", e))?;
+			if stat.clear {
+				db.do_clear_stats(stat.column.clone());
+			} else {
+				let mut out = std::io::stdout();
+				db.do_collect_stats(&mut out, stat.column.clone());
+			}
+		},
+		SubCommand::Migrate(args) => {
+			use parity_db::Options;
+			let dest_meta = Options::load_metadata(&args.dest_meta)
+				.map_err(|e| format!("Error loading dest metadata: {:?}", e))?
+				.ok_or_else(|| format!("Error opening dest metadata file"))?;
+
+			let dest_columns = dest_meta.columns;
+
+			if args.clear_dest && std::fs::metadata(&args.dest_path).is_ok() {
+				std::fs::remove_dir_all(&args.dest_path).map_err(|e| format!("Error removing dest dir: {:?}", e))?;
+			}
+			std::fs::create_dir_all(&args.dest_path).map_err(|e| format!("Error creating dest dir: {:?}", e))?;
+
+			let mut dest_options = Options::with_columns(&args.dest_path, dest_columns.len() as u8);
+			dest_options.columns = dest_columns;
+			dest_options.sync_wal = false;
+			dest_options.sync_data = false;
+
+			parity_db::migrate(&db_path, dest_options, !args.overwrite, &args.force_columns)
+				.map_err(|e| format!("Migration error: {:?}", e))?;
+
+			if args.overwrite && std::fs::metadata(&args.dest_path).is_ok() {
+				std::fs::remove_dir_all(&args.dest_path).map_err(|e| format!("Error removing dest dir: {:?}", e))?;
+			}
+		},
+		SubCommand::Check(check) => {
+			let db = parity_db::Db::open_read_only(&options)
+				.map_err(|e| format!("Invalid db: {:?}", e))?;
+			if !check.index_value {
+				// Note that we should use enum parameter instead.
+				return Err("Requires one of the following check flag: --index-value".to_string());
+			}
+			let check_param = parity_db::CheckParam::new(
+				check.column,
+				check.range_start,
+				check.range_end,
+				check.display,
+				check.display_value_max,
+			);
+			db.check_from_index(check_param)
+				.map_err(|e| format!("Check error: {:?}", e))?;
+		},
+		SubCommand::Flush(_flush) => {
+			let db = parity_db::Db::open(&options)
+				.map_err(|e| format!("Invalid db: {:?}", e))?;
+			while !db.all_empty_logs() {
```

Comment:
This should really be just `{ parity_db::Db::open() }`. It is flushed when dropped.